plugin/arcconf/arcconf.py

```diff
@@ -0,0 +1,919 @@
+                    disk_dict.update({'Channel':disk_channel, 'Device': disk_device})
+                    arcconf_disk_ids.append(disk_dict.copy())
+                    disk_dict = {}
+                    ctrl_num = cur_ctrl_num
+
+        cmds = ["create", ctrl_num, "logicaldrive", "1024", arcconf_raid_level]
+        for disk_channel_device in arcconf_disk_ids:
+            cmds.append(disk_channel_device['Channel'])
+            cmds.append(disk_channel_device['Device'])
+
+        try:
+            self._arcconf_exec(cmds, COMPONENT[5], flag_convert=False, flag_force=True)
+        except ExecError:
+            # Check whether disk is free
+            requested_disk_ids = [d.id for d in disks]
+            
+            for cur_disk in self.disks():
+                if cur_disk.id in requested_disk_ids and \
+                   not ((cur_disk.status & Disk.STATUS_FREE) or (cur_disk.status & Disk.STATUS_OK)):
+                    raise LsmError(
+                        ErrorNumber.DISK_NOT_FREE,
+                        "Disk %s is not in STATUS_FREE state" % cur_disk.id)
+            
+           
+            raise
+
+         
+        # Generate pool_id from system id and array.
+        getconfig_lds_info = self._arcconf_exec(['getconfigJSON', ctrl_num ], COMPONENT[5], False)
+        getconfig_lds_info_json = getconfig_lds_info.split("\n")[1]
+        decoded_json = json.loads(getconfig_lds_info_json)
+
+        sys_id = ctrl_num
+        cnt = ctrl_num
+
+        
+        try:
+            latest_ld = len(decoded_json['Controller']['LogicalDrive']) - 1
+            ld_num = decoded_json['Controller']['LogicalDrive'][latest_ld]['logicalDriveID']
+            ld_name = decoded_json['Controller']['LogicalDrive'][latest_ld]['name']
+            pool_id = '%s:%s' %(ctrl_num, ld_num)
+        except DeviceNotFoundError:
+            pass
+
+        lsm_vols = self.volumes(search_key='pool_id', search_value=pool_id)
+
+        if len(lsm_vols) < 1:
+            raise LsmError(
+                ErrorNumber.PLUGIN_BUG,
+                "volume_raid_create(): Got unexpected count(not 1) of new "
+                "volumes: %s" % lsm_vols)
+        return lsm_vols[0]
+
+
+
+
+    @_handle_errors
+    def volume_delete(self, volume, flags=0):
+        """
+        Depends on command:
+            arcconf delete <ctrlNo> logicaldrive <ld#> noprompt
+        """
+        if not volume.plugin_data:
+            raise LsmError(
+                ErrorNumber.INVALID_ARGUMENT,
+                "Illegal input volume argument: missing plugin_data property")
+
+        (ctrl_num, array_num, ld_num) = volume.plugin_data.split(":")
+
+        try:
+            self._arcconf_exec(['delete', ctrl_num, 'logicaldrive', array_num], COMPONENT[5], flag_convert=False, flag_force=True)
+        except ExecError:
+            raise LsmError(
+                ErrorNumber.NOT_FOUND_VOLUME,
+                "Volume not found")
```

Comment:
You should use `volumes()` to verify the existence of volume instead of treating all errors as `NOT_FOUND_VOLUME`.